File: usage/jsClient/creator/documentCreator.js
var amf = require('../../../amf-js/target/artifact/amf-module.js');
var EndPointHelper = require('./endpointHelper.js').enpointHelper;

/**
 * Client class for create a document with a web api from scratch. Use this code as example for build apis programmatically
 */
var creator = {

    spotifyApiDocument: function () {
        return new amf.Document(this.spotifyApi());
    },

    spotifyApi: function () {
        var api = new amf.WebApi();
        api.withName("Spotify Web API")
            .withVersion("v1")
            .withBasePath("{version}")
            .withContentType(["application/json"])
            .withAccepts(["application/json"]);

        EndPointHelper.populateAlbums(api.withEndPoint("/albums"));
        EndPointHelper.populateAlbumId(api.withEndPoint("/albums/{id}"));
        EndPointHelper.populateMe(api.withEndPoint("/me"));
        EndPointHelper.populatePaylists(api.withEndPoint("/me/playlists"));
        return api;
    }
}

exports.creator = creator;

File: usage/jvmClient/src/main/java/amf/example/creator/DocumentCreator.java
package amf.example.creator;

import amf.model.*;

import java.util.Collections;

/**
 * Client class for creating a document with a web api from scratch. Use this code as example for build apis programmatically
 */
public class DocumentCreator {
    
    
    
    public static Document spotifyApiDocument(){
        return new Document(spotifyApi());
    }
    
    private static WebApi spotifyApi(){
        
        WebApi api = new WebApi()
                        .withName("Spotify Web API")
                        .withVersion("v1")
                         .withBasePath("{version}")
                .withContentType(Collections.singletonList("application/json"))
                .withAccepts(Collections.singletonList("application/json"));
        
        EndPointHelper.populateAlbums(api.withEndPoint("/albums"));
        EndPointHelper.populateAlbumId(api.withEndPoint("/albums/{id}"));
        EndPointHelper.populateMe(api.withEndPoint("/me"));
        EndPointHelper.populatePaylists(api.withEndPoint("/me/playlists"));
        return api;
    }
    
    
    
    
}
